nrespipe/utils.py

```diff
@@ -570,3 +574,48 @@ def get_mag_from_simbad(target_name : str):
         mag = np.nan
 
     return mag
+
+
+def download_from_s3(frameid, output_directory):
+    url = f'{settings.ARCHIVE_FRAME_URL}/{frameid}'
+    response = requests.get(url, headers=settings.ARCHIVE_AUTH_TOKEN, stream=True).json()
+    with open(os.path.join(output_directory, response['filename']), 'wb') as f:
+        f.write(requests.get(response['url']).content)
+    return os.path.join(output_directory, response['filename'])
+
+
+def ingest_file(file_path):
+    retry = False
```

Comment:
There's a bug here. I found it because I stole your implementation :). If you set retry to false at the top, you'll never get into that while loop.

Reply:
Fixed. Thanks.